app/code/core/Mage/Catalog/Model/Category.php

```diff
@@ -45,10 +45,10 @@
  * @method $this setIncludeInMenu(int $value)
  * @method bool getInitialSetupFlag()
  * @method $this setInitialSetupFlag(bool $value)
- * @method bool getIsActive()
- * @method $this setIsActive(bool $value)
- * @method bool getIsAnchor()
- * @method $this setIsAnchor(bool $value)
+ * @method int getIsActive()
+ * @method $this setIsActive(int $value)
+ * @method int getIsAnchor()
+ * @method $this setIsAnchor(int $value)
```

Comment:
Are there going to be negative consequences of putting the return type we wish we had instead of the one we actually have? The return type is actually likely to be string "0|1" because of the horrible decision of someone long ago to make PDO return strings for all data types. But it could also be bool or int because of inconsistencies in the use of the setter. That is:

```php
$category = Mage::getModel('catalog/category')->load(1);
var_dump($category->getIsActive()); // string "0|1"
$category->setIsActive(true);
var_dump($category->getIsActive()); // bool true
```

Reply:
In the past, when i added method annotions i took the values i have found. E.g. bool, but in PDO its either string or int.  bool is wrong for DB values .

magerun uses (int) constants for active/inactive, thats looks  more correct to me. 

Id like to continue https://github.com/OpenMage/magento-lts/pull/4376 and add getter/setter methods to reflect the correct type ... later.

Reply:
Sure, just something to be thinking about. :)